src/Polly.Specs/Wrap/PolicyWrapSpecsAsync.cs

```diff
@@ -28,7 +28,7 @@ public void Nongeneric_wraps_generic_instance_syntax_wrapping_null_should_throw(
         {
             var retry = Policy.Handle<Exception>().RetryAsync(1);
 
-            Action config = () => retry.WrapAsync<int>((AsyncPolicy<int>)null);
+            Action config = () => retry.WrapAsync<int>(null);
```

Comment:
These `.WrapAsync()` methods exist in multiple forms: allowing generic-policies to wrap either non-generic or generic policies; and likewise non-generic policies to wrap either non-generic or generic policies. This is part of the flexibility of Polly `PolicyWrap`, so 'stuff just works'.

Again, tooling will suggest (absolutely correctly, from a technical perspective) that generic parameters or casts can be removed, but there is a lot of room for _human reader_ ambiguity (or having to puzzle out the meaning) once those are removed, given the number of similar overloads Polly offers.

So again: I'm preferring to retain the explicit casts to prioritise human readability.